cpp/src/common/cumlHandle.cpp

```diff
@@ -240,39 +245,35 @@ void cumlHandle_impl::waitOnInternalStreams() const {
   }
 }
 
-void cumlHandle_impl::setCommunicator( std::shared_ptr<MLCommon::cumlCommunicator> communicator )
-{
-    _communicator = communicator;
+void cumlHandle_impl::setCommunicator(
+  std::shared_ptr<MLCommon::cumlCommunicator> communicator) {
+  _communicator = communicator;
 }
 
-const MLCommon::cumlCommunicator& cumlHandle_impl::getCommunicator() const
-{
-    ASSERT(nullptr != _communicator.get(), "ERROR: Communicator was not initialized\n");
-    return *_communicator;
+const MLCommon::cumlCommunicator& cumlHandle_impl::getCommunicator() const {
+  ASSERT(nullptr != _communicator.get(),
+         "ERROR: Communicator was not initialized\n");
+  return *_communicator;
 }
 
-bool cumlHandle_impl::commsInitialized() const
-{
-    return (nullptr != _communicator.get());
+bool cumlHandle_impl::commsInitialized() const {
+  return (nullptr != _communicator.get());
 }
 
 void cumlHandle_impl::createResources() {
   cudaStream_t stream;
   CUDA_CHECK(cudaStreamCreate(&stream));
-
   CUBLAS_CHECK(cublasCreate(&_cublas_handle));
-
   CUSOLVER_CHECK(cusolverDnCreate(&_cusolverDn_handle));
-
   CUSPARSE_CHECK(cusparseCreate(&_cusparse_handle));
-
   _streams.push_back(stream);
   for (int i = 1; i < _num_streams; ++i) {
     cudaStream_t stream;
     CUDA_CHECK(cudaStreamCreate(&stream));
     _streams.push_back(stream);
   }
   CUDA_CHECK(cudaEventCreateWithFlags(&_event, cudaEventDisableTiming));
+  CUDA_CHECK(cudaGetDeviceProperties(&prop, _dev_id));
```

Comment:
Should we be worried about the cost of calling this every time we create a handle if not every caller needs it? Could we do it on demand at the first use?
(I'm ok deferring anything like that until post-0.9)